kong/db/declarative/export.lua

```diff
@@ -119,16 +119,9 @@ local function export_from_db_impl(emitter, skip_ws, skip_disabled_entities, exp
 
   local sync_version
   if emitter.want_sync_version then
-    ok, err = db.connector:query("SELECT max(version) from clustering_sync_version", "read")
-    if not ok then
-      return nil, err
-    end
+    local strategy = require("kong.clustering.services.sync.strategies.postgres").new(db)
```

Comment:
Can we initialize this once and then just reuse, e.g. a module level `get_latest_version`:

```suggestion
    if not get_latest_version then
      local strategy = require("kong.clustering.services.sync.strategies.postgres").new(db)
      get_latest_version = function()
        return strategy:get_latest_version()
      end
```

Or put it in some helper as it seems to be used in many places.

Reply:
It needs the global variable `kong.db`, I think that it can not be a module level function.

Reply:
module level function that caches it on first use

Reply:
`"kong.clustering.services.sync.strategies.postgres.new()` needs a `db` param, if it is a module level function, `kong.db` may not be initialized.

Reply:
@chronolaw I meant somethign like this:
```lua
local function get_latest_version do
  local strategy
  local function get_latest_version_real()
    return strategy:get_latest_version()
  end 
  local function get_latest_version()
    strategy = require("kong.clustering.services.sync.strategies.postgres").new(kong.db)
    get_latest_version = get_latest_version_real
    return get_latest_version()
  end
end
```

Reply:
done, many thanks.